### PR TITLE
Adds handsize to plc

### PR DESCRIPTION
**Pull request by rewanthtammana**
Signed-off-by: Rewanth Tammana <22347290+rewanthtammana@users.noreply.github.com>

This PR:

- Adds `handSize` to `PriorityLevelConfiguration` to handle low count scan job concurrency edge cases. 

### Checklist

- [x] Update changelog in CHANGELOG.md.
- [ ] Make sure `values.yaml` and `values.schema.json` are valid.
- [x] Test on WC

### CODE REVIEW

File: helm/trivy-operator/templates/priority-level-configuration.yaml [hunk 1]
```diff
@@ -8,6 +8,8 @@ spec:
     assuredConcurrencyShares: 5
     limitResponse:
       queuing:
+        # Capping handSize to 25 as queueLengthLimit is capped at 100
+        handSize: {{ min (index .Values "trivy-operator" "operator" "scanJobsConcurrentLimit") 25 }}
```
Comment (stone-z) [inline helm/trivy-operator/templates/priority-level-configuration.yaml:12]: This seems reasonable but can you explain the math here? Because I guarantee future me won't remember how this works exactly

Comment (rewanthtammana) [reply helm/trivy-operator/templates/priority-level-configuration.yaml:12]: As mentioned https://github.com/giantswarm/starboard-app/pull/99#discussion_r927635591, `handSize` can never be greater than `queueLengthLimit`. In the current configuration, `queueLengthLimit` is capped at 100 (estimating 4 calls per job).

According to https://kubernetes.io/docs/concepts/cluster-administration/flow-control/,
> A larger handSize makes it less likely for two individual flows to collide. A larger handSize also potentially increases the amount of latency that a single high-traffic flow can cause. The maximum number of queued requests possible from a single flow is handSize * queueLengthLimit.

However, a larger `handSize` also means that a small number of flows can dominate the API server, as they will be able to access more resources at the same time. To ensure `handSize` doesn't go out  of bounds in comparison with `queueLengthLimit`, the maximum value of `handSize` is set to `100/4=25`

Comment (stone-z) [reply helm/trivy-operator/templates/priority-level-configuration.yaml:12]: Thanks

**Review by stone-z — COMMENTED** [review]
Nice, let's just bump the version to v1beta2 while we're here (we'll [need to rename](https://kubernetes.io/docs/concepts/cluster-administration/flow-control/#prioritylevelconfiguration) `assuredConcurrencyShares` in v1beta3) and call this good

**Comment by rewanthtammana** [comment]
Closes https://github.com/giantswarm/giantswarm/issues/24258

**Review by stone-z — APPROVED** [review]
